src/index/index_common_param.cpp

```diff
@@ -18,14 +18,20 @@
 #include <fmt/format-inl.h>
 
 #include "common.h"
+#include "default_allocator.h"
 #include "vsag/constants.h"
 
 namespace vsag {
 
 IndexCommonParam
 IndexCommonParam::CheckAndCreate(JsonType& params, Allocator* allocator) {
     IndexCommonParam result;
-    result.allocator_ = allocator;
+
+    if (allocator != nullptr) {
+        result.allocator_ = std::make_unique<SafeAllocator>(allocator);
+    } else {
+        result.allocator_ = std::make_unique<SafeAllocator>(new DefaultAllocator(), /*owner*/ true);
+    }
```

Comment:
If we really need to make this change, we can remove the static keyword from DefaultAllocator. This part needs to be confirmed by the @wxyucs.

Reply:
@inabao you mean we don't need to make DefaultAllocator singleton anymore?

Reply:
I think singleton is not necessary

Reply:
If we create a default allocator every time we create an index, then we won't need a global singleton. This relates to the design of resources that we discussed earlier—whether we need to create a default resource each time we create an index instead of using a global resource.

> @inabao you mean we don't need to make DefaultAllocator singleton anymore?

Reply:
The management logic for `Allocator` lifecycles is scattered across many places, requiring considerable code for maintenance. I think we need more centralized and simplified logic to manage Allocators. Let's discuss this later :)